lib/Serialization/Serialization.cpp

```diff
@@ -2831,6 +2831,18 @@ void Serializer::writeDecl(const Decl *D) {
 
     auto underlying = typeAlias->getUnderlyingTypeLoc().getType();
 
+    SmallVector<TypeID, 2> dependencies;
+    for (Type dep : collectDependenciesFromType(underlying->getCanonicalType()))
+      dependencies.push_back(addTypeRef(dep));
+
+    for (Requirement req : typeAlias->getGenericRequirements()) {
```

Comment:
Do we want to factor this out into a utility function?

Reply:
Yeah, that's fair. We're not using it anywhere else yet but I bet everything else has the same problem, and I just didn't bother to check it. I'll do that in a follow-up PR.